parser-typechecker/src/Unison/Codebase/Init.hs

```diff
@@ -142,3 +145,16 @@ initCodebaseAndExit :: MonadIO m => Init m Symbol Ann -> DebugName -> Maybe Code
 initCodebaseAndExit i debugName mdir = do
   codebaseDir <- Codebase.getCodebaseDir mdir
   withNewUcmCodebaseOrExit i debugName  codebaseDir (const $ pure ())
+
+
+withTemporaryUcmCodebase ::
+  MonadUnliftIO m =>
+  Init m Symbol Ann ->
+  DebugName ->
+  ((CodebasePath, Codebase m Symbol Ann) -> m r) ->
+  m r
+withTemporaryUcmCodebase cbInit debugName action = do
```

Comment:
If folks would prefer the call site just calls `withNewUcmCodebaseOrExit` on its own that'd be fine too, it's just generally nice for things like this to clean up after themselves 😄

Reply:
It's a toss-up. Cleaning up is good (although `withSystemTempDirectory` cleans itself up somewhat), and on the other hand, having them around for debugging can be good too; similar to saving the failed transcript output.

Reply:
Yeah, the middle-ground I went with is having `transcripts` clean up after itself, if you want to debug the codebase you can use `ucm transcript` and re-run the same transcript 👍🏼